src/payment/strategies/googlepay/googlepay-braintree-initializer.ts

```diff
@@ -0,0 +1,87 @@
+import { Checkout } from '../../../checkout';
+import {
+    MissingDataError,
+    MissingDataErrorType,
+    StandardError
+} from '../../../common/error/errors';
+import PaymentMethod from '../../payment-method';
+import { BraintreeSDKCreator } from '../braintree';
+
+import {
+    GooglePaymentData,
+    GooglePayDataRequestV1,
+    GooglePayInitializer,
+    GooglePayPaymentDataRequestV1,
+    TokenizePayload
+} from './googlepay';
+import { GooglePayBraintreeSDK } from './googlepay';
+
+export default class GooglePayBraintreeInitializer implements GooglePayInitializer {
+    private _googlePaymentInstance!: GooglePayBraintreeSDK;
+
+    constructor(
+        private _braintreeSDKCreator: BraintreeSDKCreator
+    ) {}
+
+    initialize(
+               checkout: Checkout,
+               paymentMethod: PaymentMethod,
+               hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV1> {
+        if (!paymentMethod.clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        this._braintreeSDKCreator.initialize(paymentMethod.clientToken);
+
+        return this._braintreeSDKCreator.getGooglePaymentComponent()
+            .then(googleBraintreePaymentInstance => {
+                this._googlePaymentInstance = googleBraintreePaymentInstance;
+
+                return this._createGooglePayPayload(
+                    checkout,
+                    paymentMethod.initializationData.platformToken,
```

Comment:
Can you add a validation for `paymentMethod.initializationData` since you are using it here?